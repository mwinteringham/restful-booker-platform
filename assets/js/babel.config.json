{
  "presets": ["@babel/preset-env", "@babel/react"],
  "plugins": ["add-react-displayname", "@babel/plugin-syntax-dynamic-import", "@emotion"],
  "env": {
    "test": {
      "plugins": ["dynamic-import-node", "@babel/plugin-transform-runtime"]
    }
  }
}