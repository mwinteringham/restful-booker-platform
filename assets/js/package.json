{
  "name": "ui",
  "version": "1.6.0",
  "description": "React based UI interface for Restful-Booker-Platform",
  "scripts": {
    "test": "jest",
    "test:watch": "jest --watchAll",
    "build": "webpack --mode=production",
    "dev": "webpack-dev-server --mode=development"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/mwinteringham/restful-booker-platform.git"
  },
  "author": "Mark Winteringham",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/mwinteringham/restful-booker-platform/issues"
  },
  "homepage": "https://github.com/mwinteringham/restful-booker-platform#readme",
  "devDependencies": {
    "@babel/core": "^7.20.2",
    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
    "@babel/plugin-transform-runtime": "^7.19.6",
    "@babel/preset-env": "^7.20.2",
    "@babel/preset-react": "^7.18.6",
    "@emotion/babel-plugin": "^11.10.5",
    "@emotion/styled": "^11.10.5",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "axios": "^1.1.3",
    "babel-core": "^6.26.3",
    "babel-jest": "^29.3.1",
    "babel-loader": "^9.1.0",
    "babel-plugin-add-react-displayname": "0.0.5",
    "babel-plugin-dynamic-import-node": "^2.3.3",
    "css-loader": "^6.7.2",
    "debug": "~4.3.4",
    "emotion": "^11.0.0",
    "file-loader": "^6.2.0",
    "history": "^5.3.0",
    "html-loader": "^4.2.0",
    "html-webpack-plugin": "^5.5.0",
    "identity-obj-proxy": "^3.0.0",
    "isomorphic-fetch": "^3.0.0",
    "jest": "^29.3.1",
    "moment": "^2.29.4",
    "moment-timezone": "^0.5.40",
    "morgan": "^1.10.0",
    "nock": "^13.2.9",
    "node-sass": "^8.0.0",
    "pigeon-maps": "^0.21.3",
    "pigeon-marker": "^0.3.4",
    "query-string": "^7.1.1",
    "react": "^18.2.0",
    "react-big-calendar": "^1.5.2",
    "react-datepicker": "^4.8.0",
    "react-dom": "^18.2.0",
    "react-modal": "^3.16.1",
    "react-router-dom": "^6.4.3",
    "react-spinners": "^0.13.6",
    "react-tooltip": "^4.5.0",
    "reactjs-popup": "^2.0.5",
    "sass-loader": "^13.2.0",
    "style-loader": "^3.3.1",
    "universal-cookie": "^4.0.4",
    "url-loader": "^4.1.1",
    "validate.js": "^0.13.1",
    "webpack": "^5.75.0",
    "webpack-cli": "^5.0.0",
    "webpack-dev-server": "^4.11.1"
  },
  "jest": {
    "testMatch": [
      "**/tests/**/*-test.js?(x)",
      "**/?(*.)(spec|test).js?(x)"
    ],
    "moduleNameMapper": {
      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
      "\\.(css|less)$": "identity-obj-proxy"
    },
    "testEnvironment": "jsdom"
  },
  "dependencies": {
    "jest-environment-jsdom": "^29.3.1"
  }
}
