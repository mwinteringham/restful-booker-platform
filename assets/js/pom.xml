<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>com.automationintesting</groupId>
    <artifactId>restful-booker-platform-assets-js</artifactId>
    <version>1.5.${revision}</version>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <revision>SNAPSHOT</revision>
    </properties>

    <build>
        <plugins>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>exec-maven-plugin</artifactId>
                <version>1.3.2</version>
                <executions>
                    <execution>
                        <id>npm install</id>
                        <goals>
                            <goal>exec</goal>
                        </goals>
                        <phase>initialize</phase>
                        <configuration>
                            <executable>npm</executable>
                            <arguments>
                                <argument>install</argument>
                            </arguments>
                        </configuration>
                    </execution>

                    <execution>
                        <id>npm run test</id>
                        <goals>
                            <goal>exec</goal>
                        </goals>
                        <phase>test</phase>
                        <configuration>
                            <executable>npm</executable>
                            <arguments>
                                <argument>run</argument>
                                <argument>test</argument>
                            </arguments>
                        </configuration>
                    </execution>

                    <execution>
                        <id>npm run build</id>
                        <goals>
                            <goal>exec</goal>
                        </goals>
                        <phase>install</phase>
                        <configuration>
                            <executable>npm</executable>
                            <arguments>
                                <argument>run</argument>
                                <argument>build</argument>
                            </arguments>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>

</project>