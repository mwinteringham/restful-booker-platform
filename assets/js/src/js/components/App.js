import React, { useEffect, useState } from 'react';
import { HashRouter, Route, Routes } from 'react-router-dom'
import ReactModal from 'react-modal';
import Cookies from 'universal-cookie';

const AdminContainer = React.lazy(() => import('./AdminContainer.js'));
const CookiePolicy = React.lazy(() => import('./CookiePolicy.js'));
const PrivacyPolicy = React.lazy(() => import('./PrivacyPolicy.js'));

import Banner from './Banner.js';
import Home from './Home.js';
import Footer from './Footer.js';

import Loading from './Loading.js';

if (process.env.NODE_ENV !== 'test') ReactModal.setAppElement('#root');

const App = () => {
    
    const [showBanner, toggleBanner] = useState(false);

    useEffect(() => {
        const cookies = new Cookies();
    
        if(typeof cookies.get('banner') === 'undefined'){
            toggleBanner(true);
        }
    }, [])

    let welcome = null;

    if(showBanner){
        welcome = <div>
            <Banner setWelcome={toggleBanner}/>
        </div>
    }

    return(
        <HashRouter>
            {welcome}
                <Routes>
                    <Route path='/admin/*' element={<React.Suspense fallback={<Loading />}><AdminContainer /></React.Suspense>} />
                    <Route exact path='/' element={<Home />} />
                    <Route exact path='/cookie' element={<React.Suspense fallback={<Loading />}><CookiePolicy /></React.Suspense>} />
                    <Route exact path='/privacy' element={<React.Suspense fallback={<Loading />}><PrivacyPolicy /></React.Suspense>} />
                </Routes>
            <Footer />
        </HashRouter>
    );
}

export default App;
