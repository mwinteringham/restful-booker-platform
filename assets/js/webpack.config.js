const path = require("path");
const HtmlWebPackPlugin = require("html-webpack-plugin");

const PATHS = {
  build: path.join(__dirname, '..', 'api', 'target', 'classes', 'public')
};

module.exports = {
  devServer: {
    compress: false,
    port: 3003
  },
  entry: ["./src/js/index.js"],
  output: {
    path: PATHS.build,
    filename: "js/[name].js"
  },
  module: {
    rules: [
      { test: /\.js$/, exclude: /node_modules/, loader: "babel-loader"},
      { test: /\.jsx$/, loader: 'babel-loader', exclude: /node_modules/ },
      { test: /\.html$/, loader: "html-loader" },
      { test: /\.css$/, use: ['style-loader', 'css-loader'] },
      { test: /\.scss$/, use: ['style-loader', 'css-loader', 'sass-loader'] },
      { test: /\.(gif|png|jp(e*)g|svg)$/,  use: [{
        loader: 'url-loader',
        options: { 
            limit: 8000, // Convert images < 8kb to base64 strings
            name: 'images/[hash]-[name].[ext]'
        } 
      }]}
    ]
  },
  plugins: [
    new HtmlWebPackPlugin({
      template: "./src/index.html",
      filename: "./index.html"
    })
  ]
};