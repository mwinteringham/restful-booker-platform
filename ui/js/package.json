{
  "name": "ui",
  "version": "1.0.0",
  "description": "React based UI interface",
  "scripts": {
    "test": "jest",
    "test:watch": "jest --watchAll",
    "build": "webpack --mode=production",
    "dev": "webpack-dev-server --mode=development --port 3003"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/mwinteringham/restful-booker-platform.git"
  },
  "author": "Mark Winteringham",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/mwinteringham/restful-booker-platform/issues"
  },
  "homepage": "https://github.com/mwinteringham/restful-booker-platform#readme",
  "devDependencies": {
    "@babel/core": "^7.3.4",
    "@babel/plugin-syntax-dynamic-import": "^7.2.0",
    "@babel/preset-env": "^7.3.4",
    "@babel/preset-react": "^7.0.0",
    "@emotion/styled": "^10.0.9",
    "babel-core": "^7.0.0-bridge.0",
    "babel-loader": "^8.0.5",
    "babel-plugin-add-react-displayname": "0.0.5",
    "babel-plugin-dynamic-import-node": "^2.2.0",
    "css-loader": "^2.1.1",
    "debug": "~4.1.1",
    "emotion": "^10.0.9",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.11.2",
    "enzyme-to-json": "^3.3.5",
    "file-loader": "^3.0.1",
    "html-loader": "^0.5.5",
    "html-webpack-plugin": "^3.2.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^24.5.0",
    "moment": "^2.24.0",
    "morgan": "^1.9.1",
    "nock": "^10.0.6",
    "node-fetch": "^2.3.0",
    "node-sass": "^4.11.0",
    "pigeon-maps": "^0.11.11",
    "pigeon-marker": "^0.3.4",
    "query-string": "^6.4.0",
    "react": "^16.8.4",
    "react-calendar-heatmap": "^1.8.0",
    "react-datepicker": "^2.2.0",
    "react-dom": "^16.8.4",
    "react-router-dom": "^4.4.0",
    "react-spinners": "^0.5.3",
    "react-tooltip": "^3.10.0",
    "reactjs-popup": "^1.3.2",
    "sass-loader": "^7.1.0",
    "style-loader": "^0.23.1",
    "universal-cookie": "^3.0.7",
    "url-loader": "^1.1.2",
    "validate.js": "^0.12.0",
    "webpack": "^4.23.1",
    "webpack-cli": "^3.3.0",
    "webpack-dev-server": "^3.2.1"
  },
  "jest": {
    "testMatch": [
      "**/tests/**/*-test.js?(x)",
      "**/?(*.)(spec|test).js?(x)"
    ],
    "setupFiles": [
      "./tests/jestsetup.js"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "moduleNameMapper": {
      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
      "\\.(css|less)$": "identity-obj-proxy"
    }
  },
  "dependencies": {
    "react-modal": "^3.8.1"
  }
}
