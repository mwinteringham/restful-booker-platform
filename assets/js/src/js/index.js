import React from 'react';
import {createRoot} from 'react-dom/client';
import App from './components/App.js';
import '../styles/style.scss';

const rootElement = document.getElementById('root');
const root = createRoot(rootElement);

root.render(
  <App />
)
