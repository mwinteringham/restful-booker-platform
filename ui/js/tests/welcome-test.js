import React from 'react';
import Welcome from '../src/js/components/Welcome.jsx';
import App from '../src/js/components/App.jsx';
import nock from 'nock';
import { mountWrap } from './helpers/routerWrapper.js';

beforeEach(() => {
    nock('http://localhost:3004')
        .post('/auth/validate')
        .reply(200);
});

test('Welcome component contains correct details', () => {
    const welcomeComponent = shallow(
        <Welcome />
    )

    expect(welcomeComponent).toMatchSnapshot();
});

test('Welcome component appears on initial visit', () => {
    const appComponent = shallow(
        <App />
    )

    appComponent.setState({
        showWelcome : true
    })

    expect(appComponent).toMatchSnapshot();
});

test('Welcome component doesnt appear after close', () => {
    const appComponent = shallow(
        <App />
    )

    appComponent.setState({
        showWelcome : false
    })

    expect(appComponent).toMatchSnapshot();
});
