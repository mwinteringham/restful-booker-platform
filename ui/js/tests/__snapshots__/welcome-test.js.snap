// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clicking on close button closes modal 1`] = `
<App>
  <div>
    <Switch />
    <Footer>
      <footer
        className="footer"
        id="footer"
      >
        <div
          className="container"
        >
          <p
            className="text-muted"
          >
            restful-booker-platform v4.0 - Created by 
            <a
              href="http://www.mwtestconsultancy.co.uk"
            >
              Mark Winteringham
            </a>
             / 
            <a
              href="https://thefriendlytester.co.uk/"
            >
              Richard Bradshaw
            </a>
             - Â© 2019 
            <a
              href="/#/cookie"
            >
              Cookie-Policy
            </a>
             - 
            <a
              href="/#/privacy"
            >
              Privacy-Policy
            </a>
             - 
            <a
              href="/#/admin"
            >
              Admin panel
            </a>
             
            <span
              style={
                Object {
                  "float": "right",
                }
              }
            >
              Learn more about 
              <a
                alt="Automation in testing site"
                href="https://automationintesting.com"
              >
                Automation in Testing
              </a>
               here
            </span>
          </p>
        </div>
      </footer>
    </Footer>
  </div>
</App>
`;

exports[`Welcome component appears on initial visit 1`] = `
<div>
  <div>
    <Welcome
      setWelcome={[Function]}
    />
  </div>
  <Switch>
    <Route
      path="/admin/"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/cookie"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/privacy"
    />
  </Switch>
  <Footer />
</div>
`;

exports[`Welcome component contains correct details 1`] = `
<Popup
  arrow={true}
  arrowStyle={Object {}}
  className=""
  closeOnDocumentClick={false}
  closeOnEscape={false}
  contentStyle={Object {}}
  defaultOpen={true}
  disabled={false}
  keepTooltipInside={false}
  lockScroll={false}
  modal={true}
  mouseEnterDelay={100}
  mouseLeaveDelay={100}
  offsetX={0}
  offsetY={0}
  on={
    Array [
      "click",
    ]
  }
  onClose={[Function]}
  onOpen={[Function]}
  open={false}
  overlayStyle={Object {}}
  position="bottom center"
  repositionOnResize={true}
  trigger={null}
>
  <div
    className="modal-body welcome"
  >
    <div
      className="container-fluid"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-1"
        />
        <div
          className="col-sm-10 content"
        >
          <br />
          <br />
          <h2>
            Welcome to Restful Booker Platform
          </h2>
          <br />
          <br />
          <h4>
            Your one stop shop to practise Software Testing!
          </h4>
        </div>
        <div
          className="col-sm-1"
        />
      </div>
      <div
        className="row justify-content-center"
      >
        <div
          className="col-sm-3"
        />
        <div>
          <div
            className="btn-group mr-2"
            role="group"
          >
            <button
              className="btn btn-outline-primary disabled"
              id="prev"
              onClick={[Function]}
            >
              Prev
            </button>
            <button
              className="btn btn-outline-primary"
              id="next"
              onClick={[Function]}
            >
              Next
            </button>
          </div>
        </div>
        <div
          className="col-sm-3"
        />
      </div>
    </div>
  </div>
</Popup>
`;

exports[`Welcome component doesnt appear after close 1`] = `
<div>
  <Switch>
    <Route
      path="/admin/"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/cookie"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/privacy"
    />
  </Switch>
  <Footer />
</div>
`;
